crates/script/src/receipts.rs

```diff
@@ -71,31 +71,54 @@ pub async fn check_tx_status(
 pub fn format_receipt(chain: Chain, receipt: &AnyTransactionReceipt) -> String {
     let gas_used = receipt.gas_used;
     let gas_price = receipt.effective_gas_price;
-    format!(
-        "\n##### {chain}\n{status} Hash: {tx_hash:?}{caddr}\nBlock: {bn}\n{gas}\n\n",
-        status = if !receipt.inner.inner.inner.receipt.status.coerce_status() {
-            "❌  [Failed]"
-        } else {
-            "✅  [Success]"
-        },
-        tx_hash = receipt.transaction_hash,
-        caddr = if let Some(addr) = &receipt.contract_address {
-            format!("\nContract Address: {}", addr.to_checksum(None))
-        } else {
-            String::new()
-        },
-        bn = receipt.block_number.unwrap_or_default(),
-        gas = if gas_price == 0 {
-            format!("Gas Used: {gas_used}")
-        } else {
-            let paid = format_units(gas_used.saturating_mul(gas_price), 18)
-                .unwrap_or_else(|_| "N/A".into());
-            let gas_price = format_units(U256::from(gas_price), 9).unwrap_or_else(|_| "N/A".into());
-            format!(
-                "Paid: {} ETH ({gas_used} gas * {} gwei)",
-                paid.trim_end_matches('0'),
-                gas_price.trim_end_matches('0').trim_end_matches('.')
-            )
-        },
-    )
+
+    if shell::is_json() {
+        let _ = sh_println!(
+            "{}",
+            serde_json::json!({
+                "chain": chain,
+                "status": if !receipt.inner.inner.inner.receipt.status.coerce_status() {
```

Comment:
maybe we should have vars like
`let success = receipt.inner.inner.inner.receipt.status.coerce_status()` defined before `if shell::is_json() ` and used for both branches in order to avoid duplication? (same for `block_number` and `contract_address` which doesn't seem to be checksumed for json)